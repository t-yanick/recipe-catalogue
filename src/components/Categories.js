import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCategoriesStartAsync } from '../actions/getCategories';

const Categories = (props) => {
  const categories = useSelector(state => state.categories.categories);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCategoriesStartAsync());
  });

  const handleChange = e => {
    props.onCategoryChange(e.target.value);
  };

  return (
    <div>
      <h3>Select a category</h3>
      <select className="category" placeholder="Select Category" onChange={handleChange}>
        {categories && categories.map(cat => (
          <option key={cat.idCategory} value={cat.strCategory}>
            {cat.strCategory}
          </option>
        ))}
      </select>
    </div>
  );
};

Categories.propTypes = {
  onCategoryChange: PropTypes.func,
};

Categories.defaultProps = {
  onCategoryChange: null,
}

export default Categories;