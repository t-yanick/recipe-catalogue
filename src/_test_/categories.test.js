import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Categories from '../components/Categories';

describe('Creates a component', () => {
  const middlewares = [thunk];
  const initialState = { categories: [{ cat: 'lamb' }] };
  const mockStore = configureStore(middlewares);
  let store;
  it('renders correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(
      <Provider store={store}>
        <Categories />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a component', () => {
    store = mockStore(initialState);
    render(<Provider store={store}><Categories /></Provider>);
    const element = screen.getByText('Select a category');
    expect(element).toBeInTheDocument();
  });
});
