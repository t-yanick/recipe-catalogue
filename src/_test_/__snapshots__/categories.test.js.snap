// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component renders correctly 1`] = `
<div>
  <h3
    className="mt-2"
  >
    Select a category
  </h3>
  <select
    className="category"
    onChange={[Function]}
    placeholder="Select Category"
  />
</div>
`;
